rest/server_context.go

```diff
@@ -1131,16 +1131,27 @@ func initClusterAgent(clusterAddress, clusterUser, clusterPass, certPath, keyPat
 	return agent, nil
 }
 
-// initializeGoCBAgent Obtains a gocb agent from the current server connection. Requires the agent to be closed after use
+// initializeGoCBAgent Obtains a gocb agent from the current server connection. Requires the agent to be closed after use.
+// Uses retry loop
 func (sc *ServerContext) initializeGoCBAgent() (*gocbcore.Agent, error) {
-	agent, err := initClusterAgent(
-		sc.config.Bootstrap.Server, sc.config.Bootstrap.Username, sc.config.Bootstrap.Password,
-		sc.config.Bootstrap.X509CertPath, sc.config.Bootstrap.X509KeyPath, sc.config.Bootstrap.CACertPath, sc.config.Bootstrap.ServerTLSSkipVerify,
-		sc.config.API.ServerReadTimeout.Value())
+	err, a := base.RetryLoop("Initialize Cluster Agent", func() (shouldRetry bool, err error, value interface{}) {
+		agent, err := initClusterAgent(
```

Comment:
other issues inside `initClusterAgent`:
- leaking gocb agent under error scenarios, need to close an agent if it's not being returned